crates/cargo-wdk/src/actions/build.rs

```diff
@@ -52,72 +56,60 @@ impl<'a> BuildAction<'a> {
-        match manifest_path.strip_extended_length_path_prefix() {
-            Ok(path) => Ok(Self {
-                package_name,
-                profile,
-                verbosity_level,
-                manifest_path: path,
-                command_exec,
-            }),
-            Err(StripExtendedPathPrefixError::NoExtendedPathPrefix) => Ok(Self {
-                package_name,
-                profile,
-                verbosity_level,
-                manifest_path,
-                command_exec,
-            }),
+        let mut manifest_path = fs_provider.canonicalize_path(&working_dir.join("Cargo.toml"))?;
+        manifest_path = match manifest_path.strip_extended_length_path_prefix() {
+            Ok(path) => path,
+            Err(StripExtendedPathPrefixError::NoExtendedPathPrefix) => manifest_path,
             Err(StripExtendedPathPrefixError::EmptyPath) => {
-                Err(BuildActionError::EmptyManifestPath)
+                return Err(BuildActionError::EmptyManifestPath);
             }
-        }
+        };
+        Ok(Self {
+            package_name,
+            profile,
+            target_arch,
+            verbosity_level,
+            manifest_path,
+            command_exec,
+        })
     }
 
     /// Entry point method to run the build action
     /// # Returns
-    /// * `Result<(), CommandError>` - Result indicating success or failure of
-    ///   the build action
+    /// * `Result<(), BuildActionError>` - Result indicating success or failure
+    ///   of the build action
     /// # Errors
     /// * `CommandError` - If the command execution fails
     pub fn run(&self) -> Result<(), BuildActionError> {
-        info!(
-            "Running cargo build for package: {}, profile: {}",
-            self.package_name, self.profile
-        );
-        let manifest_path = self.manifest_path.to_string_lossy().to_string();
-        let profile = &self.profile.to_string();
-        let args = trace::get_cargo_verbose_flags(self.verbosity_level).map_or_else(
-            || {
-                vec![
-                    "build",
-                    "--manifest-path",
-                    &manifest_path,
-                    "-p",
-                    self.package_name,
-                    "--profile",
-                    profile,
-                ]
-            },
-            |flag| {
-                vec![
-                    "build",
-                    flag,
-                    "--manifest-path",
-                    &manifest_path,
-                    "-p",
-                    self.package_name,
-                    "--profile",
-                    profile,
-                ]
-            },
-        );
-
+        info!("Running cargo build for package: {}", self.package_name);
+        let mut args = vec!["build".to_string()];
+        args.push("-p".to_string());
+        args.push(self.package_name.to_string());
+        if let Some(path) = self.manifest_path.to_str() {
```

Comment:
Since manifest_path is required argument, we have to handle else case for this here.